### PR TITLE
Make milling results definable in JSON (#41174)

### PR DESCRIPTION
**Pull request by KheirFerrum**
#### Summary

SUMMARY: Infrastructure "Ports over jsonized milling results"

#### Purpose of change

Port over [41174](https://github.com/CleverRaven/Cataclysm-DDA/pull/41174) from DDA repo that allows one to define items as millable and define their results outside of hardcode.

Converted their implementation to use int values so as to prevent products being made out of thin air.

#### Describe the solution

> Removes the (undocumented) flag "MILLABLE".
> 
> Adds an item slot that defines what an item can be milled into (e.g. for "wheat" that would be "flour").
> 
> Fixes mills to produces flour at the same conversion rate as manual crafting. Mills used to give 15 flour for 1 input item. But crafting it with a quern requires different amounts of input items to get the same 15 flour as result. For example it takes four oats to craft 15 flour. This has been fixed (80 oats give only 300 flout, 80 wheat give 1200 flour).
> 
> Fixes the product to have the same properties as if crafted by hand (inheriting flags and components). This makes flour milled from oats contain different vitamins than when milled from wheat, just like you would get if you had crafted it manually.
> 
> Makes chitin pieces and bone (human as well) millable. The result is chitin powder and bone meal.
> 
> Fixes a crash (see commit).

#### Additional context

In the future we'd probably want scaling changes to time based on materials in the mill, and if at all possible have it process in increments, so that stopping the mill halfway has half of the millable items changed to products. But this system is one step in the right direction for now.

### CODE REVIEW

**Comment by Secret-Cisco** [comment]
What still needs to be done to take this out of draft?

**Comment by KheirFerrum** [comment]
> What still needs to be done to take this out of draft?

I pretty much just need to test it so that it works as intended, I'd normally do it when publishing the PR but I was busy driving and rushing other responsibilities.

File: doc/JSON_INFO.md [hunk 1]
```diff
@@ -1635,6 +1635,10 @@ See also VEHICLE_JSON.md
     [ "9mm", [ "glockmag" ] ]                // The first magazine specified for each ammo type is the default
     [ "45", [ "m1911mag", "m1911bigmag" ] ],
 ],
+"milling": {                                 // Optional. If given, the item can be milled in a water/wind mill.
+  "into": "flour",                           // The item id of the result of the milling.
+  "conversion_rate": 1.0                     // Conversion of number of items that are milled (e.g. with a rate of 2, 10 input items will yield 20 milled items).
```
Comment (olanti-p) [inline doc/JSON_INFO.md:1640]: The docs still imply floating-point numbers are supported.
Also IMO, could use a better wording.

**Comment by Coolthulhu** [comment]
Tiny issue: the "you need to wait more" message assumes grains, so it's weird if you insert bones instead.

**Pull request merged**